packages/database/src/DatabaseDialectInitializer.php

```diff
@@ -3,14 +3,21 @@
 namespace Tempest\Database;
 
 use Tempest\Container\Container;
-use Tempest\Container\Initializer;
+use Tempest\Container\DynamicInitializer;
 use Tempest\Database\Config\DatabaseConfig;
 use Tempest\Database\Config\DatabaseDialect;
+use Tempest\Reflection\ClassReflector;
+use UnitEnum;
 
-final class DatabaseDialectInitializer implements Initializer
+final class DatabaseDialectInitializer implements DynamicInitializer
```

Comment:
Refactored to `DynamicInitializer`, because there can now be multiple `DatabaseDialect` instances, one per registered database connection

Reply:
Is this class really needed though? Why not expose a `dialect` property on the `Database` interface?

Reply:
True, I found it convenient being able to inject `DatabaseDialect` directly in some places. But it's not strictly necessary.